lib/parse.js

```diff
@@ -207,6 +208,10 @@ var normalizeParseOptions = function normalizeParseOptions(opts) {
         return defaults;
     }
 
+    if (opts.allowEmptyArrays && typeof opts.allowEmptyArrays !== 'boolean') {
+        throw new TypeError('allowEmptyArrays option property can only be true or false');
+    }
```

Comment:
Please add parse and stringify tests that ensure it throws for anything that's not true, false, or undefined.